ir/src/main/scala/org/scalajs/ir/Serializers.scala

```diff
@@ -1232,6 +1293,56 @@ object Serializers {
     def readStrings(): List[String] =
       List.fill(readInt())(readString())
 
+    private def readLocalName(): LocalName = {
+      val i = readInt()
+      val s = strings(i)
+      if (!isStringAlreadyValidated(i, ValidatedAsLocalNameBit)) {
+        newLocalName(s)
+        didValidateString(i, ValidatedAsLocalNameBit)
+      }
+      s.asInstanceOf[LocalName]
+    }
+
+    private def readLabelName(): LabelName = {
+      val i = readInt()
+      val s = strings(i)
+      if (!isStringAlreadyValidated(i, ValidatedAsLabelNameBit)) {
+        newLabelName(s)
+        didValidateString(i, ValidatedAsLabelNameBit)
+      }
+      s.asInstanceOf[LabelName]
+    }
+
+    private def readFieldName(): FieldName = {
+      val i = readInt()
+      val s = strings(i)
+      if (!isStringAlreadyValidated(i, ValidatedAsFieldNameBit)) {
+        newFieldName(s)
+        didValidateString(i, ValidatedAsFieldNameBit)
+      }
+      s.asInstanceOf[FieldName]
+    }
+
+    private def readMethodName(): MethodName = {
+      val i = readInt()
+      val s = strings(i)
+      if (!isStringAlreadyValidated(i, ValidatedAsMethodNameBit)) {
+        newMethodName(s)
+        didValidateString(i, ValidatedAsMethodNameBit)
+      }
+      s.asInstanceOf[MethodName]
+    }
+
+    private def readClassName(): ClassName = {
+      val i = readInt()
+      val s = strings(i)
+      if (!isStringAlreadyValidated(i, ValidatedAsClassNameBit)) {
+        newClassName(s)
+        didValidateString(i, ValidatedAsClassNameBit)
+      }
+      s.asInstanceOf[ClassName]
+    }
+
```

Comment:
The idea is that names are at least internally consistent, always. The IR checker validates class structures and type relations, which need some global information, but the structure of names can be validated one by one.

This will become critical for method names in the next (and last) refactoring in the series, because they will have actual structure, which can only *exist* if the encoded name is validated.